### PR TITLE
feat: Adds trimBeforeSubstring custom func

### PR DESCRIPTION
**Pull request by joshmyers**
## What
This commit adds a custom template func to trim anything before a sub string match. Also adds a few built ins from the strings package to allow for easier mangling of the output via custom templates.

## Why
We have a number of projects/stacks that bundle assets along with CDK code, things like Docker images for custom resource functions and friends.

This bundling happens during a CDK synth/diff and causes a whole load of non CDK diff related output to be shown, before any diffs. This isn't helpful and pushes the comment length unnecessarily closer to the truncate limit.

image

### CODE REVIEW

**Comment by karlderkaefer** [comment]
Thank you for contributing, I will check within next 3 days

File: transform/template.go [hunk 1]
```diff
@@ -161,7 +162,13 @@ func (ct *commentTemplate) getCustomTemplate() (string, error) {
 func (t *commentTemplate) render() (string, error) {
 	templateContent := t.ChooseTemplate().getTemplateContent()
 	logrus.Debugf("Using template content %s", templateContent)
-	tmpl, err := template.New("commentTemplate").Parse(templateContent)
+	tmpl, err := template.New("commentTemplate").Funcs(template.FuncMap{
```
[the remark this answers is not in the document]
Comment (joshmyers) [reply transform/template.go]: @karlderkaefer thanks for the quick review.

Adding Sprig would definitely add flexibility and as you say, offload additional docs around custom functions to the lib. I've added a commit to this PR, let me know what you think. I can rebase/fixup however required.

tl;dr I couldn't  _quite_ get what I needed out of sprig alone, as it's index function is not for Strings and splitting / joining `.Content` drops color output.

**Review by karlderkaefer — APPROVED** [review]
Looks good

**Comment by joshmyers** [comment]
@karlderkaefer I don't have write access to merge.
